packages/story-editor/src/components/canvas/useInsertElement.js

```diff
@@ -98,6 +132,7 @@ function useInsertElement() {
       if (resource) {
         backfillResource(resource);
         handleRegisterUsage(resource);
+        generateBaseColorExternal(element);
```

Comment:
Hm, the only issue here is with adding incorrect history entries if the element is changed _after_ insertion and not at the same time.

Pascal created [an issue](https://github.com/google/web-stories-wp/issues/9577) to move the transient data out of the `resource` object, could we perhaps not store the `baseColor` there either anymore to avoid history issues? Or is there any other alternative for not corrupting history? Thoughts?

An example issue it can create:
- User inserts an image
- Two history entires get created: the insertion and adding the base color
- User clicks "Undo" once
- The image stays but the base color is removed, nothing changes visibly for the user.

Reply:
That issue is for transient state like `isMuting` though, which we only need to know about for a couple of seconds. But `baseColor` is something that will need to be stored **persistently**.

Where else would we store it if not on the resource?

> Or is there any other alternative for not corrupting history?

I was hoping you had some ideas :-)

I see three options:

1. Generate base color **async** after insertion like in this PR -> corrupts history
2. Generate base color **sync** during/before insertion -> not ideal UX
3. Generate base color **async** after insertion, but don't create a new history entry and just add it to the current history entry. -> better UX, no corrupt history  
  Kinda what you mentioned [here](https://github.com/google/web-stories-wp/issues/9550#issuecomment-953828865)

As for (2), you mentioned [alternative options](https://github.com/google/web-stories-wp/issues/9550#issuecomment-953801633) like calculating on hover, but that doesn't seem feasible as it's not super fast. So no guarantee that it works.

IMO (3) would be most compelling: Having a way to update an element/story/page/resource without adding a new history entry.

I think such a functionality would be useful beyond this single use case.

Reply:
> As for (2), you mentioned alternative options like calculating on hover, but that doesn't seem feasible as it's not super fast. So no guarantee that it works.

We could potentially wait with insertion until the calculation that started on hovering has been completed. However, agreed that (3) would be the best option.

Perhaps we can change the logic of history so that it would generally allow ignoring specific props as needed. This would need a separate issue though for the history logic change, so that this PR could just define the variable to ignore. cc @barklund since you implemented the original history tracking.

Is it acceptable to corrupt the history for now within this PR, or should this new history change be a blocker for the current PR?

Reply:
My vote is for 3. 

If there a way to add a param to `updateElementById` that would skip the adding to history. Something like this. 

```
updateElementById({ elementId: id, properties: { ...changes, ignoreHistory: true } });
```

Reply:
Added an issue for it: https://github.com/google/web-stories-wp/issues/9645

There's one issue that would come with this approach as well (editor would not indicate unsaved state), added details to the description.

Reply:
It is also worth noting, this generation of base color on insertion like this, is only done on hotlink / external resources. Base Color is generated once and cache for assets stored / cached. So even if merge this, it will only be a issue for external resources.

Reply:
> Is it acceptable to corrupt the history for now within this PR

Yes IMHO. Soon this will be a rare scenario, because in the future all media items will have a base color during upload.